app/services/report_importer.rb

```diff
@@ -46,13 +47,10 @@ def import
     host.save(:validate => false)
 
     # and save our report
-    @report = Report.new(:host => host, :reported_at => time, :status => st, :metrics => raw['metrics'])
-    return report unless report.save
-    # Store all Puppet message logs
-    import_log_messages
-    # Check for errors
-    inspect_report
+    @report = report_name_class.new(:host => host, :reported_at => time, :status => status, :metrics => raw['metrics'])
+    return report unless @report.save
     logger.info("Imported report for #{name} in #{(Time.now - start_time).round(2)} seconds")
```

Comment:
I'm not entirely sure what to suggest, but we've lost the timing information around the import_log_messages method.